fe/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java

```diff
@@ -373,4 +390,59 @@ public static long analyzeTimeout(Map<String, String> properties, long defaultTi
         }
         return timeout;
     }
+
+    public static Map<String, String> analyzeDynamicPartition(Database db, OlapTable olapTable,
+                                                              Map<String, String> properties) throws DdlException {
+        if (properties == null || properties.isEmpty()) {
+            throw new DdlException("Table " + olapTable.getName() + " is not a dynamic partition olap table");
+        }
+        Map<String, String> analyzedProperties = new HashMap<>();
+        if (properties.containsKey(PROPERTIES_DYNAMIC_PARTITION_TIME_UNIT)) {
+            String timeUnit = properties.get(PROPERTIES_DYNAMIC_PARTITION_TIME_UNIT);
+            DynamicPartitionUtils.checkTimeUnit(olapTable, timeUnit);
```

Comment:
How about move the function to DynamicPartitionUtils

Reply:
> In my opinion, it's not a good idea to check property in PropertyAnalyzer. It would make this class will reference too many class.

What about move the function to DynamicPartitionUtils?